### PR TITLE
Prune API of the resolver package.

### PR DESCRIPTION
**Pull request by benluddy**
The resolver package exports a number of identifiers that have not been used for resolution in over a year. As part of the ongoing effort to decouple purely-resolution-related concerns from the rest of OLM, some of these have been rehomed into the packages that still consume them, and entirely unused code has been removed.

### CODE REVIEW

File: test/e2e/subscription_e2e_test.go
Comment (dinhxuanvu) [inline test/e2e/subscription_e2e_test.go:221]: I know this is a test so it doesn't matter but this const `SkipPackageAnnotationKey` still exists. May as well keeping using it instead of hardcoding.

File: pkg/controller/operators/catalog/operator.go
Comment (njhale) [inline pkg/controller/operators/catalog/operator.go:893]: TIL: this wasn't (never?) actually dependency injected

even if it was, the DI should have taken a factory anyway

File: pkg/controller/operators/catalog/querier.go
Comment (dinhxuanvu) [inline pkg/controller/operators/catalog/querier.go:52]: I kinda wish we get rid of this as well. This func call is just to detect if there is a new version by making multiple API calls to registry to query the info. It doesn't really matter to the resolution which is run later if there is a change. Sometimes, `FindReplacement` will find something that will replace existing version and change the Subscriptions status to `UpgradeAvailable`. However, when the resolver runs, the replacement may not meet the constraints so no upgrade happens. That's a bit confusing. Plus, may as well let the resolver run because maybe it is less expensive (API-call-wise) that way. That part is unknown so it is fine to keep it as it is.

**Review by dinhxuanvu — COMMENTED** [review]
Looking good. Just a tiny nit.
I can lgtm if it is good to go. Certainly don't want to hold this on a tiny nit.

**Review by njhale — APPROVED** [review]
At times like this I'm glad we haven't bumped the module to v1 yet :)

/lgtm

File: test/e2e/subscription_e2e_test.go
Comment (benluddy) [reply test/e2e/subscription_e2e_test.go:221]: I decided that the end-to-end test shouldn't depend on this constant from the implementation. If its value is changed, the test should fail and need to be updated since it represents a change in behavior.

File: pkg/controller/operators/catalog/querier.go
Comment (benluddy) [reply pkg/controller/operators/catalog/querier.go:52]: I feel the same way, but I didn't want to take on that change right now.

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Pull request merged**
